Google-Summer-of-Code-2025.md

```diff
@@ -1221,52 +1259,211 @@ This project entails introducing a developer-only UI (as part of the developer o
+
+**Product/technical clarifiers:** @BenHenning (product + technical)
+
+**Required knowledge/skills:**
+- Figure out the root cause of an issue and communicate it well using a [debugging doc](https://github.com/oppia/oppia/wiki/Debugging-Docs).
+- Build the app and install it on a local device or emulator. Then verify that you can (a) build a non-test library  target locally, (b) run a unit test locally, and (c) play through the app locally.
+- Write new Kotlin code with unit tests.
+
+**Related issues:**
+
+Issues related to portions of the codebase that will be affected by this project:
+- Most issues under the 'priority' section of the [CLaM team board](https://github.com/orgs/oppia/projects/4/views/3) are likely to help build familiarity with broad parts of the codebase which will help with addressing lint issues.
+- Adding tests for scripts will help build familiarity with the scripting work: [#4971](https://github.com/oppia/oppia-android/issues/4971). The stats script is especially relevant, too, as it is built to wrap utilities normally used via the command line (not unlike Android lint), including AAPT and apkanalyzer.
+
+**Suggested Milestones:**
+- **Milestone 1**: Key deliverables:
+  - Introduce a new script that can run Android lint checks with support for a local textproto file that can override specific failures to allow them to pass.
+  - Check in a populated exemption list with all known exemptions.
+  - Introduce a new wiki page explaining how to use the script and update the exemption list.
+  - Introduce CI support for the new script so that it runs for every change to develop and for all PRs.
+
+- **Milestone 2**: Key deliverables:
+  - Fix all of the error categories of lint issues, plus the warning categories mentioned in the project description.
+  - Ensure all exemptions that aren't fixed as part of this project have corresponding tracking issues filed and spec'd with enough context for contributors to work on them.
+
+<details>
+<summary>Org-admin/tech-lead commentary/advice</summary>
+
+Scripts usually seem like straightforward changes, and they often can be. However, they almost always uncover unexpected complexities that need to be solved mid-project (which is why this is considered a more difficult project overall). It's also the case that many of the lint issues that are included as part of this project may not have known solutions ahead of time which requires additional problem solving that may occur after the design phase of the project.
+
+That being said, there are many examples of both scripts and exemptions to reference when designing and constructing the script itself. While the exact functionality being implementing (wrapping the Android linter and creating an intermediary step between the exemption XML and a textproto version of it) isn't well defined, the other boilerplate of a script (textproto loading, script library, running scripts in CI, etc.) is well defined and referenceable.
```

Comment:
Second sentence might be wrong ("While the exact functionality being implementing") -- should it be "implemented"?

Perhaps say: `isn't well defined yet (and should be explained clearly in your proposal)`?

Reply:
Yes, I also found this awkward but couldn't think of anything better late last night. Your wording clears up the bit that was odd with the previous wording.